modules/manage/partials/iceberg/query-iceberg-topics.adoc

```diff
@@ -0,0 +1,142 @@
+| `redpanda-iceberg-catalog/redpanda/<topic-name>`
+
+|===
+endif::[]
+
+If your engine needs the full JSON metadata path, use the following:
+
+```
+redpanda-iceberg-catalog/metadata/redpanda/<topic-name>/v<version-number>.metadata.json
+```
+
+This provides read access to all snapshots written as of the specified table version (denoted by `version-number`).
+
+NOTE: Redpanda automatically removes expired snapshots on a periodic basis. Snapshot expiry helps maintain a smaller metadata size and reduces the window available for <<time-travel-queries,time travel>>.
+
+== Query examples
+
+ifndef::env-byoc[]
+To follow along with the examples on this page, suppose you produce the same stream of events to a topic `ClickEvent`, which uses a schema, and another topic `ClickEvent_key_value`, which uses the key-value mode. The topics have xref:manage:tiered-storage.adoc[Tiered Storage] configured to an AWS S3 bucket. A sample record contains the following data:
+endif::[]
+
+ifdef::env-byoc[]
+To follow along with the examples on this page, suppose you produce the same stream of events to a topic `ClickEvent`, which uses a schema, and another topic `ClickEvent_key_value`, which uses the key-value mode. The topic's Iceberg data is stored in an AWS S3 bucket. A sample record contains the following data:
+endif::[]
+
+[,bash,role=no-copy]
+----
+{"user_id": 2324, "event_type": "BUTTON_CLICK", "ts": "2024-11-25T20:23:59.380Z"}
+----
+
+=== Topic with schema (`value_schema_id_prefix` mode)
+
+In this example, it is assumed you have created the `ClickEvent` topic, set `redpanda.iceberg.mode` to `value_schema_id_prefix`, and are connecting to a REST-based Iceberg catalog. The following is an Avro schema for `ClickEvent`:
+
+.`schema.avsc`
+[,avro]
+----
+{
+    "type" : "record",
+    "namespace" : "com.redpanda.examples.avro",
+    "name" : "ClickEvent",
+    "fields" : [
+       { "name": "user_id", "type" : "int" },
+       { "name": "event_type", "type" : "string" },
+       { "name": "ts", "type": "string" }
+    ]
+ }
+----
+
+. Register the schema under the `ClickEvent-value` subject:
++
+[,bash]
+----
+rpk registry schema create ClickEvent-value --schema path/to/schema.avsc --type avro
+----
+
+. Produce to the `ClickEvent` topic using the following format:
++
+[,bash]
+----
+echo '"key1" {"user_id":2324,"event_type":"BUTTON_CLICK","ts":"2024-11-25T20:23:59.380Z"}' | rpk topic produce ClickEvent --format='%k %v\n' --schema-id=topic
+----
++
+The `value_schema_id_prefix` requires that you produce to a topic using the Schema Registry wire format, which includes the magic byte and schema ID in the prefix of the message payload. This allows Redpanda to identify the correct schema version in the Schema Registry for a record. See the https://www.redpanda.com/blog/schema-registry-kafka-streaming#how-does-serialization-work-with-schema-registry-in-kafka[Understanding Apache Kafka Schema Registry^] blog post to learn more about the wire format.
+
```

Comment:
The link goes to a subsection of the blog, but maybe the whole sentence should change. I searched the blog for "wire format" and "wire" and got nothing.

Reply:
Updated - still mention wire format, but not explicitly linking it to the referenced blog post